File: JAVA/AdvanceContracepticPlan.java
//DO
//Ask user for last period date, cycle length, and period duration
//Calculate:
//- Next period date
//- Ovulation date
//- Fertile window (ovulation ± 2 days)
//- Safe periods (after period and before next period)

//Display all results
//Ask if user wants to repeat
//WHILE user says 'Y'
//Display closing message

import java.util.Scanner;
public class AdvanceContracepticPlan {
 public static void main(String[] args) {
  Scanner input = new Scanner(System.in);
int day;
int month;
int year;
int cycleLength;
int periodDuration;
char repeat;

do {
System.out.println("Menstrual Cycle plan ");
System.out.print("Enter last period day (1-31): ");
day = input.nextInt();

 System.out.print("Enter last period month (1-12): ");
 month = input.nextInt();
 System.out.print("Enter last period year : ");
 year = input.nextInt();
 System.out.print("Enter cycle length in days: ");
 
 cycleLength = input.nextInt();
 System.out.print("Enter period duration in days: ");
 periodDuration = input.nextInt();
 int nextDay = day + cycleLength;
 int nextMonth = month;
 int nextYear = year;

while (nextDay > 30) {
nextDay -= 30;
nextMonth++;
if (nextMonth > 12) {
nextMonth = 1;
nextYear++;}
}

int ovulationDay = day + (cycleLength / 2);
int ovulationMonth = month;
int ovulationYear = year;
while (ovulationDay > 30) {
ovulationDay -= 30;
ovulationMonth++;
if (ovulationMonth > 12) {
ovulationMonth = 1;
ovulationYear++;
}
}

int fertileStart = ovulationDay - 2;
int fertileEnd = ovulationDay + 2;

System.out.println("Next Period: " + nextDay + "/" + nextMonth + "/" + nextYear);
System.out.println("Ovulation Date: " + ovulationDay + "/" + ovulationMonth + "/" + ovulationYear);
System.out.println("Fertile Window: " + fertileStart + " to " + fertileEnd );
System.out.println("Safe Periods:");
System.out.println("  After Period: Day " + (day + periodDuration) + " to " + (fertileStart - 1));
System.out.println("  Before Next Period: Day " + (fertileEnd + 1) + " to " + (nextDay - 1));
System.out.print("\nDo you want to calculate again? (Y/N): ");
repeat = input.next().charAt(0);
} while (repeat == 'Y' || repeat == 'y');

System.out.println("Your consultation has ended. Stay sharp,\nMay we not be victim of parent of unwanted pregnancy\nDr. Adedeji Fareed Cares!");
}
}

File: python/AdvanceContracepticPlan.py
def advance_contraceptic_plan():
    recap = 'yes'

    while recap.lower().trim() == 'yes':
        print("Enter your preferred Menstrual Cycle plan")

        day = int(input("Enter last period day (1-31): "))
        month = int(input("Enter last period month (1-12): "))
        year = int(input("Enter last period year: "))
        cycle_length = int(input("Enter cycle length in days: "))
        period_duration = int(input("Enter period duration in days: "))

       
        next_day = day + cycle_length
        next_month = month
        next_year = year

        while next_day > 30:
            next_day -= 30
            next_month += 1
            if next_month > 12:
                next_month = 1
                next_year += 1

        ovulation_day = day + (cycle_length / 2)
        ovulation_month = month
        ovulation_year = year

        while ovulation_day > 30:
            ovulation_day -= 30
            ovulation_month += 1
            if ovulation_month > 12:
                ovulation_month = 1
                ovulation_year += 1

        fertile_start = ovulation_day - 2
        fertile_end = ovulation_day + 2

    
        print(f"Next Period: {next_day}/{next_month}/{next_year}")
        print(f"Ovulation Date: {ovulation_day}/{ovulation_month}/{ovulation_year}")
        print(f"Fertile Window: {fertile_start} to {fertile_end}")
        print("Safe Periods:")
        print(f"  After Period: Day {day + period_duration} to {fertile_start - 1}")
        print(f"  Before Next Period: Day {fertile_end + 1} to {next_day - 1}")

        recap = input("\nDo you want to calculate again? (Y/N): ")

    print("\nYour consultation has ended. Stay sharp,")
    print("May we not be victims of unwanted pregnancy.")
    print("Dr. Adedeji Fareed Cares!")
advance_contraceptic_plan()